src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaSubmissionSelectionService.java

```diff
@@ -0,0 +1,113 @@
+package de.tum.in.www1.artemis.service.connectors.athena;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.TextSubmission;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.service.dto.athena.TextExerciseDTO;
+
+/**
+ * Service for selecting the "best" submission to assess right now using Athena, e.g. by the highest information gain.
+ * Assumes that submissions have already been sent to Athena (it only sends submission IDs to choose from).
+ * The default choice if Athena does not respond is to choose a random submission.
+ */
+@Service
+@Profile("athena")
+public class AthenaSubmissionSelectionService {
+
+    // pretty short timeout, because this should be fast, and it's not too bad if it fails
+    private static final int REQUEST_TIMEOUT_MS = 1000;
+
+    private final Logger log = LoggerFactory.getLogger(AthenaSubmissionSelectionService.class);
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
+
+    private final AthenaConnector<RequestDTO, ResponseDTO> connector;
+
+    private static class RequestDTO {
+
+        public TextExerciseDTO exercise;
+
+        public List<Long> submissionIds; // Athena just needs submission IDs => quicker request, because less data is sent
+
+        RequestDTO(@NotNull TextExercise exercise, @NotNull List<Long> submissionIds) {
+            this.exercise = TextExerciseDTO.of(exercise);
+            this.submissionIds = submissionIds;
+        }
+    }
+
+    private record ResponseDTO(long data // will contain the submission ID to choose, or -1 if no submission was explicitly chosen
```

Comment:
```suggestion
    private record ResponseDTO(long submissionId // will contain the submission ID to choose, or -1 if no submission was explicitly chosen
```

Reply:
Good suggestion! I initially thought I had to use this design because Athena sends a module response wrapped in a JSON object with `data` as the actual contents, but now I discovered `@JsonProperty("data") long submissionId` and use it.